kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/spec/SpecRunner.kt

```diff
@@ -43,6 +44,24 @@ abstract class SpecRunner(val listener: TestEngineListener) {
          Try { listener.specInstantiationError(kclass, it) }
       }
 
+   protected suspend fun run(
+      concurrencyMode: ConcurrencyMode,
+      testCases: Collection<TestCase>,
+      run: suspend (TestCase) -> Unit
+   ) = testCases.forEach { run(it) }
+//      when (concurrencyMode) {
+//         ConcurrencyMode.Isolated -> testCases.forEach { run(it) }
+//         ConcurrencyMode.Concurrent -> coroutineScope {
```

Comment:
Enabling this would provide huge value for slow network and timeout bound test suites 🙏 

Suggestions for naming: ConcurrencyMode.[None, Coarse, Fine] or ConcurrencyMode.[None, Spec, Test]

Reply:
I've updated this now.